src/Xrefcheck/Verify.hs

```diff
@@ -511,88 +523,52 @@ verifyReference
       VerifyResult [ExternalHttpTooManyRequests retryAfter] -> retryAfter
       _ -> Nothing
 
-    isVirtual = matchesGlobPatterns root (ecIgnoreLocalRefsTo cExclusions)
+    isVirtual canonicalRoot = matchesGlobPatterns canonicalRoot (ecIgnoreLocalRefsTo cExclusions)
 
-    checkRef mAnchor referredFile = verifying $
-      unless (isVirtual referredFile) do
-        checkReferredFileIsInsideRepo referredFile
-        mFileStatus <- tryGetFileStatus referredFile
+    checkRef mAnchor referredFile canonicalRoot = verifying $
+      unless (isVirtual canonicalRoot referredFile) do
+        referredFileRelative <- case stripPrefix
```

Comment:
This allows to both check if the file is inside the repo and to obtain the relative path of the file inside the repo, so it can be used to print them nicely.

Reply:
Interesting, let's go with it.